server/src/routines/turn-actions.ts

```diff
@@ -29,6 +29,7 @@ function getAttack(
 	if (!creator.isHermit()) return []
 
 	const nextAttack = creator.props.getAttack(game, creator, hermitAttackType)
+	creator.player.updateLastUsedHermitAttack(hermitAttackType)
```

Comment:
This is the only place that the selected attack type can be forwarded when the attack action is executed. In order for Amnesia to block Puppetry/Role Play when they mock a primary, attack type must read here instead of `attack.type`. Alternative hooks to `beforeAttack` are not usable because attacks can be canceled after running `getAttackRequests` and `getAttack` does not take a `HermitAttackType` parameter

Reply:
That makes sense. I don't really think this is a refactor worthy thing to worry about so I'm happy with this.